scripts/test/hlint.nix

```diff
@@ -0,0 +1,36 @@
+{ runCommand, hlint, src, lib }:
+
+let
+  cleanSourceFilter = with lib;
+    name: type: let baseName = baseNameOf (toString name); in (
+      (type == "regular" && hasSuffix ".hs" baseName) ||
+      (type == "regular" && hasSuffix ".yaml" baseName) ||
+      (type == "directory")
+    );
+  src' = builtins.filterSource cleanSourceFilter src;
```

Comment:
Can this use `lib.cleanSourceWith` (see example in #3174). That will work better with already filtered sources I think.